src/ServiceFabric/ServiceFabric/Commands/ServiceFabricClusterCertificateCmdlet.cs

```diff
@@ -97,6 +97,11 @@ public abstract class ServiceFabricClusterCertificateCmdlet : ServiceFabricClust
         [ValidateNotNullOrEmpty]
         public string SecretIdentifier { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = ByExistingKeyVault,
```

Comment:
>ParameterSetName = ByExistingKeyVault, [](start = 64, length = 38)

This looks a bit odd to me - both the SecretIdentifier and the Thumbprint parameter are:
- validate not null or empty
- corresponding to the same parameter set - byExistingKV

Either of them would be necessary and sufficient, but not both? Or did you mean to create a new parameter set?

Reply:
validate not null or empty is a check whenever the parameters are used. I'm using the same parameter set and only having SecretIdentifier as mandatory since Thumbprint is only needed for the cases where we can't get the certificate bundle form the key vault. Only using Thumbprint parameter is not enough, that's why SecretIdentifier  is mandatory

---
In reply to: [418234139](https://github.com/Azure/azure-powershell/pull/11758#discussion_r418234139) [](ancestors = 418234139)